pkg/operation/botanist/addons.go

```diff
@@ -172,6 +168,60 @@ func (b *Botanist) DeployManagedResources(ctx context.Context) error {
 		}
 	}
 
+	// Create the cloud config managed resource that contains:
+	// 1. A secret containing the dedicated cloud config execution script for each worker group
+	// 2. A secret containing some shared RBAC policies for downloading the cloud config execution script
+	cloudConfigCharts := map[string]func() (*chartrenderer.RenderedChart, error){
+		"shoot-cloud-config-rbac": b.generateCloudConfigRBACChart,
+	}
+	bootstrapTokenSecret, err := kutil.ComputeBootstrapToken(ctx, b.K8sShootClient.Client(), utils.ComputeSHA256Hex([]byte(time.Now().Format("2006-01-02")))[:6], "A bootstrap token generated by Gardener.", 48*time.Hour)
+	if err != nil {
+		return fmt.Errorf("error computing bootstrap token for shoot cloud config: %+v", err)
+	}
+	//  for each worker pool add a secret containing the cloud config execution script
+	for _, worker := range b.Shoot.Info.Spec.Provider.Workers {
+		name := fmt.Sprintf("shoot-cloud-config-execution-%s", worker.Name)
+		cloudConfigCharts[name] = b.getGenerateCloudConfigExecutionChartFunc(name, worker, bootstrapTokenSecret)
+	}
+	cloudConfigManagedResource := manager.
+		NewManagedResource(b.K8sSeedClient.Client()).
+		WithNamespacedName(b.Shoot.SeedNamespace, "shoot-cloud-config-execution").
+		WithLabels(labels).
+		WithInjectedLabels(injectedLabels).
+		KeepObjects(false)
+
+	// reference secrets
+	for name, renderChartFunc := range cloudConfigCharts {
+		renderedChart, err := renderChartFunc()
+		if err != nil {
+			return fmt.Errorf("error rendering %q chart: %+v", name, err)
+		}
+		data := renderedChart.AsSecretData()
+
+		secretName := "managedresource-" + name
+		if err := manager.
+			NewSecret(b.K8sSeedClient.Client()).
+			WithNamespacedName(b.Shoot.SeedNamespace, secretName).
+			WithKeyValues(data).
+			Reconcile(ctx); err != nil {
+			return err
+		}
+		cloudConfigManagedResource.WithSecretRef(secretName)
+	}
+	if err := cloudConfigManagedResource.Reconcile(ctx); err != nil {
+		return err
+	}
+	// TODO: remove this in the future
+	return b.deleteLegacyCloudConfigSecret()
+}
+
+// deleteLegacyCloudConfigSecret deletes the legacy cloud-config-execution secret created by gardenlet prior to v1.3.0
+func (b *Botanist) deleteLegacyCloudConfigSecret() error {
+	secretName := "managedresource-shoot-cloud-config-execution"
+	err := b.K8sSeedClient.Kubernetes().CoreV1().Secrets(b.Shoot.SeedNamespace).Delete(secretName, metav1.NewDeleteOptions(0))
```

Comment:
Any specific reason, why you set `deletionGracePeriodSeconds=0`?

Reply:
There is not specific reason actually, I will change it to unset

Reply:
Addressed in https://github.com/gardener/gardener/commit/b8aadd8ebe56c6a226687d3f640295056f05c25c